stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -104,4 +112,8 @@ def _run_split_procurement(self, procurements):
                 getattr(self.env["stock.rule"], "_run_%s" % rule.mto_rule_id.action)(
                     [(mto_procurement, rule.mto_rule_id)]
                 )
+            done_procurements_by_product.setdefault(procurement.product_id.id, 0)
+            done_procurements_by_product[
+                procurement.product_id.id
+            ] += procurement.product_qty
```

Comment:
Good point, this needs to be fixed.
@ferran-S73 Could you attend this comment ?

Reply:
@jbaudoux I don't think I fully understand what you're trying to say. I think the case you're talking about is already tested in the test code I wrote.
If we have 1 unit in stock and 2 procurements for 2 units each, we need to resupply a total of 3 units. If we only take into account mts quantity for the second procurement it will only resupply 1 unit, thus only making a total of 2 units.

Reply:
@jbaudoux maybe I'm not so used to this module but I don't know how you could decide to not split between mto & mts. As far as I'm concerned the expected behavior would always be for the first procurement to take the available mts quantity.

Reply:
@jbaudoux I understand that, buy I don't see how would you do it using this module since (correct me if I'm wrong) it doesn't allow you to choose if you wish to use the on hand quantity or not